src/nodes.coffee

```diff
@@ -1219,12 +1220,22 @@ exports.Obj = class Obj extends Base
         key  = key.properties[0].name
         prop = new Assign key, prop, 'object'
       if key is prop
-        if prop.shouldCache()
-          [key, value] = prop.base.cache o
-          key  = new PropertyName key.value if key instanceof IdentifierLiteral
-          prop = new Assign key, value, 'object'
-        else if not prop.bareLiteral?(IdentifierLiteral)
-          prop = new Assign prop, prop, 'object'
+        unless propHasSplat
+          if prop.shouldCache()
+            [key, value] = prop.base.cache o
+            key  = new PropertyName key.value if key instanceof IdentifierLiteral
+            prop = new Assign key, value, 'object'
+          else unless prop.bareLiteral?(IdentifierLiteral)
+            prop = new Assign prop, prop, 'object'
+        else
+          # Spread in CSX.
+          prop = if @csx then new Literal "{#{prop.compile(o)}}" else prop
+      # Check if CSX attribute is valid.
+      if @csx and prop instanceof Assign
+        prop.variable.error "Unexpected token" unless prop.variable.unwrap() instanceof PropertyName
+        propVal = prop.value.unwrap()
+        unless ((propVal instanceof Parens or propVal instanceof StringWithInterpolations) and propVal.body instanceof Block) or propVal instanceof StringLiteral or propVal instanceof Obj
```

Comment:
Could we reformat this line to something like:

```coffee
unless propVal instanceof StringLiteral or propVal instanceof Obj or
    ((propVal instanceof Parens or propVal instanceof StringWithInterpolations) and propVal.body instanceof Block)
  prop.value.error "..."
```

It might also benefit from a comment stating what is allowed (e.g. what is `(propVal instanceof Parens or propVal instanceof StringWithInterpolations) and propVal.body instanceof Block` ensuring?).